developer-docs-site/docs/move/book/functions.md

```diff
@@ -521,44 +521,58 @@ fun foo() { return () }
 
 ## Inline Functions
 
-Inline functions are functions which are expanded at caller side instead
-of compiled into Move bytecode. This allows to save gas and may lead
-to faster execution. For example, one can define an inline function
+Inline functions are functions whose bodies are expanded in-place at the caller location during compile time.
+Thus, inline functions do not appear in Move bytecode as a separate function: all calls to them are expanded away by the compiler.
+They allow to save gas and may lead to faster execution.
+However, users should be aware that they could lead to larger bytecode size.
+
+One can define an inline function by adding the `inline` keyword to a function declaration as shown below:
 
 ```move=
 inline fun percent(x: u64, y: u64):u64 { x * 100 / y }
 ```
 
-Now, when call `percent(2, 200)` the compiler will inline the function
-definition as if the user has written `2 * 100 / 200`.
+If we call this inline function as `percent(2, 200)`, the compiler will replace this call with the inline function's body, as if the user has written `2 * 100 / 200`.
+
+### Function parameters and lambda expressions
 
-### Function Parameters
+Inline functions support _function parameters_, which accept lambda expressions (i.e., higher-order functions) as arguments.
+This feature allows writing several common programming patterns elegantly.
+Similar to inline functions, lambda expressions are also expanded at call site.
 
-Inline functions support _function parameters_. This allows
-to define higher-order functions in Move which can comprehend common
-programming patterns. As inline functions are expanded at compilation time,
-this feature of function parameters can be supported without direct
-support for it in Move bytecode.
+A lambda expression includes a list of parameter names (enclosed within `||`) followed by the body.
+Some simple examples are: `|x| x + 1`, `|x, y| x + y`, `|| 1`, `|| { 1 }`.
+A lambda's body can refer to variables available in the scope where the lambda is defined: this is also known as capturing.
```

Comment:
Such variables can be read or written by the lambda (if mutable); since a lambda expression currently can only occur as a direct argument to a call to an inline function, there are no complicated scoping issues as in some other languages.

Reply:
I am assuming you meant add this text here. Let me know if not.